src/control/cmd/daos/pool.go

```diff
@@ -276,6 +278,100 @@ func (cmd *poolQueryCmd) Execute(_ []string) error {
 	return nil
 }
 
+type poolQueryTargetsCmd struct {
+	poolBaseCmd
+
+	Rank      uint32 `long:"rank" required:"1" description:"Engine rank of the targets to be queried"`
+	Targetidx string `long:"target-idx" description:"Comma-separated list of target idx(s) to be queried"`
+}
+
+func convertDaosSpaceInfo(in *C.struct_daos_space, mt C.uint) *mgmtpb.StorageTargetUsage {
+	if in == nil {
+		return nil
+	}
+
+	return &mgmtpb.StorageTargetUsage{
+		Total: uint64(in.s_total[mt]),
+		Free:  uint64(in.s_free[mt]),
+	}
+}
+
+// For using the pretty printer that dmg uses for this target info.
+//
+// Does the same thing as ds_mgmt_drpc_pool_query_target() will do, to stuff the target info
+// into a protobuf message, then use the automatic conversion from proto to control.
+
+func convertPoolTargetInfo(ptinfo *C.daos_target_info_t) (*control.PoolQueryTargetInfo, error) {
+	pqtp := new(mgmtpb.PoolQueryTargetInfo)
+
+	// daos_target_info_t -> mgmtpb
+	pqtp.Type = mgmtpb.PoolQueryTargetInfo_TargetType(ptinfo.ta_type)
+	pqtp.State = mgmtpb.PoolQueryTargetInfo_TargetState(ptinfo.ta_state)
+	pqtp.Perf = &mgmtpb.TargetPerf{Foo: int32(ptinfo.ta_perf.foo)}
+	pqtp.Space = []*mgmtpb.StorageTargetUsage{
+		convertDaosSpaceInfo(&ptinfo.ta_space, C.DAOS_MEDIA_SCM),
+		convertDaosSpaceInfo(&ptinfo.ta_space, C.DAOS_MEDIA_NVME),
+	}
+
+	// Manual conversion mgmtpb -> control; convert.Types not working for some reason (type, state)
+	pqti := new(control.PoolQueryTargetInfo)
+	pqti.Type = control.PoolQueryTargetType(pqtp.Type)
+	pqti.State = control.PoolQueryTargetState(pqtp.State)
+	pqti.Perf = &control.TargetPerf{Foo: pqtp.Perf.Foo}
+	pqti.Space = []*control.StorageTargetUsage{
+		{pqtp.Space[uint(C.DAOS_MEDIA_SCM)].Total, pqtp.Space[uint(C.DAOS_MEDIA_SCM)].Free},
+		{pqtp.Space[uint(C.DAOS_MEDIA_NVME)].Total, pqtp.Space[uint(C.DAOS_MEDIA_NVME)].Free},
+	}
+
+	//return pqti, convert.Types(pqtp, pqti)
+	return pqti, nil
+}
+
+func (cmd *poolQueryTargetsCmd) Execute(_ []string) error {
+	cleanup, err := cmd.resolveAndConnect(C.DAOS_PC_RO, nil)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	var idxlist []uint32
+	if err = common.ParseNumberList(cmd.Targetidx, &idxlist); err != nil {
+		return errors.WithMessage(err, "parsing target list")
+	}
+
+	var inforesp *control.PoolQueryTargetResp = &control.PoolQueryTargetResp{}
+	var ptinfo C.daos_target_info_t
+	var rc C.int
```

Comment:
A few notes regarding style/idioms:
It's a minor thing, but in our codebase we use the `camelCase` (unexported symbol) or `CamelCase` (exported symbol) conventions for names.

Also, in Go, it's usually considered clearer to use the `new()` keyword to initialize an empty struct. It's also more idiomatic to use the `:=` operator to declare and initialize a variable, e.g. `infoResp := new(control.PoolQueryTargetResp)`. The way you have things now is not wrong, it just looks a little odd.